migrate/secrets.go

```diff
@@ -2,10 +2,96 @@ package migrate
 
 import (
 	"database/sql"
+	"encoding/json"
+	"fmt"
+
+	"github.com/russross/meddler"
+	"github.com/sirupsen/logrus"
 )
 
 // MigrateSecrets migrates the secrets V0 database
 // to the V1 database.
 func MigrateSecrets(source, target *sql.DB) error {
-	return nil
+	secretsV0 := []*SecretV0{}
+
+	if err := meddler.QueryAll(source, &secretsV0, secretImportQuery); err != nil {
+		return err
+	}
+
+	logrus.Infof("migrating %d secrets", len(secretsV0))
+	tx, err := target.Begin()
+
+	if err != nil {
+		return err
+	}
+
+	defer tx.Rollback()
+
+	for _, secretV0 := range secretsV0 {
+		log := logrus.WithFields(logrus.Fields{
+			"repo":   secretV0.RepoFullname,
+			"secret": secretV0.Name,
+		})
+
+		log.Debugln("migrate secret")
+
+		repoV1 := &RepoV1{}
+
+		if err := meddler.QueryRow(target, repoV1, fmt.Sprintf(repoSlugQuery, secretV0.RepoFullname)); err != nil {
```

Comment:
I believe you can search by `sercretV0.RepoID` since we preserve the ID

Reply:
At least for my MariaDB test databases I had the case if mismatch with the ids...